traits/tests/test_date.py

```diff
@@ -62,7 +71,7 @@ def test_assign_non_date(self):
         with self.assertRaises(TraitError) as exception_context:
             obj.simple_date = "1975-2-13"
         message = str(exception_context.exception)
-        self.assertIn("must be a date or None", message)
+        self.assertIn("must be a date or None, but", message)
```

Comment:
consistency fix: in a later test, we need to check for "must be a date, but" instead of checking for "must be a date", to avoid an incorrect test pass if the error message is actually "must be a date or None". So we introduce the ", but" part of the check here for consistency.